server/api/src/main/java/org/hawkular/apm/server/api/task/ProcessingUnit.java

```diff
@@ -106,7 +106,12 @@ public void handle(String tenantId, List<T> items) throws Exception {
         List<T> retries = null;
         RetryAttemptException lastException = null;
 
-        processor.initialise(tenantId, items);
+        try {
+            processor.initialise(tenantId, items);
+        } catch (RetryAttemptException e) {
+            retries = items;
```

Comment:
How does the retry handler distinguish "failure in initialization + failure in handling all items" from "success in initialization + failure in handling all items"?